pandas/io/excel/_base.py

```diff
@@ -1326,6 +1326,11 @@ def _value_with_fmt(
             fmt = "0"
         else:
             val = str(val)
+            # GH#56954
+            if len(val) > 32767:
+                raise ValueError(
```

Comment:
Can you add a reference where `32767` comes from?

Also I think this should be a `UserWarning` and not a `ValueError`. We shouldn't completely stop writing the excel file due to excel's limitation

Reply:
I had suggested an error over a warning in https://github.com/pandas-dev/pandas/pull/57103#discussion_r1468876833 primarily due to the fact that was raise an error if the frame is too long or wide already.

Reply:
Ah I see. The reason I suggested warning was due to a similar scenario I remember a while back where we used to raise an error if we detected whether a `stack/unstack` operation would create "too many elements" and users wanted to bypass that limitation. We ended up changing it to a `PerformanceWarning` https://github.com/pandas-dev/pandas/pull/45084

I slightly prefer to still raise a `UserWarning` to make pandas less opinionated here (the Excel output is still valid and maybe a truncated value is OK for some users). But if you feel strongly about a `ValueError`, it's not a blocker for me

Reply:
Actually - raising here would need a deprecation too, right? That makes me think we should raise a warning. @luke396 - can you turn this back into a warning?

Reply:
> Actually - raising here would need a deprecation too, right?

Conservatively yes, but it would "fix" an undesired (external) behavior ("bug").